src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCommon.java

```diff
@@ -0,0 +1,294 @@
+        addColumnMetadataInternal(positionInSource, name, jdbcType, precision, scale, dateTimeFormatter);
+    }
+
+    /**
+     * Adds metadata for the given column in the batch/file.
+     * 
+     * @param positionInSource
+     *            Indicates which column the metadata is for. Columns start at 1.
+     * @param name
+     *            Name for the column (optional if only using column ordinal in a mapping for SQLServerBulkCopy operation)
+     * @param jdbcType
+     *            JDBC data type of the column
+     * @param precision
+     *            Precision for the column (ignored for the appropriate data types)
+     * @param scale
+     *            Scale for the column (ignored for the appropriate data types)
+     * @throws SQLServerException
+     *             when an error occurs
+     */
+    public void addColumnMetadata(int positionInSource,
+            String name,
+            int jdbcType,
+            int precision,
+            int scale) throws SQLServerException {
+        addColumnMetadataInternal(positionInSource, name, jdbcType, precision, scale, null);
+    }
+
+    /**
+     * Adds metadata for the given column in the batch/file.
+     * 
+     * @param positionInSource
+     *            Indicates which column the metadata is for. Columns start at 1.
+     * @param name
+     *            Name for the column (optional if only using column ordinal in a mapping for SQLServerBulkCopy operation)
+     * @param jdbcType
+     *            JDBC data type of the column
+     * @param precision
+     *            Precision for the column (ignored for the appropriate data types)
+     * @param scale
+     *            Scale for the column (ignored for the appropriate data types)
+     * @param dateTimeFormatter
+     *            format to parse data that is sent
+     * @throws SQLServerException
+     *             when an error occurs
+     */
+    void addColumnMetadataInternal(int positionInSource,
+            String name,
+            int jdbcType,
+            int precision,
+            int scale,
+            DateTimeFormatter dateTimeFormatter) throws SQLServerException {
+        loggerExternal.entering(loggerClassName, "addColumnMetadata", new Object[] {positionInSource, name, jdbcType, precision, scale});
+
+        String colName = "";
+
+        if (0 >= positionInSource) {
+            MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_invalidColumnOrdinal"));
+            Object[] msgArgs = {positionInSource};
+            throw new SQLServerException(form.format(msgArgs), SQLState.COL_NOT_FOUND, DriverError.NOT_SET, null);
+        }
+
+        if (null != name)
+            colName = name.trim();
+        else if ((columnNames != null) && (columnNames.length >= positionInSource))
+            colName = columnNames[positionInSource - 1];
+
+        if ((columnNames != null) && (positionInSource > columnNames.length)) {
```

Comment:
just nitpicking but seems a bit inconsistent here  as other places we usually do (null != blah)
same for lines 157, 159

Reply:
Sure. fixed.